pkg/cache/cache.go

```diff
@@ -398,23 +398,23 @@ func (c *Cache) DeleteClusterQueue(cq *kueue.ClusterQueue) {
 	metrics.ClearCacheMetrics(cq.Name)
 }
 
-func (c *Cache) AddCohort(apiCohort *kueuealpha.Cohort) {
+func (c *Cache) AddOrUpdateCohort(apiCohort *kueuealpha.Cohort) {
 	c.Lock()
 	defer c.Unlock()
-	cohort := newCohort(apiCohort.Name)
-	c.hm.AddCohort(cohort)
+	c.hm.AddCohort(apiCohort.Name)
+	cohort := c.hm.Cohorts[apiCohort.Name]
```

Comment:
Isn't it risky to allow to access `Cohorts` map directly, not using a method? Especially if we have add and delete methods for it.

Reply:
I agree, but this is outside of the scope of this PR.

When implementing hierarchy.Manager, I left it as a public field to make the change as pain free as possible. The further complicating factor is that we test against those maps in the cache/snapshot tests

Opened https://github.com/kubernetes-sigs/kueue/issues/2996